acs-integration-tests/src/test/java/com/ge/predix/test/utils/ACSRestTemplateFactory.java

```diff
@@ -154,12 +156,26 @@ public OAuth2RestTemplate getOAuth2ResttemplateForZAC() {
         return this.zacTemplate;
     }
 
+    // TODO Move the UAA template gets to UAATestUtil so that the ACSRestTemplateFactory is only for ACSRestTemplates
     public OAuth2RestTemplate getOAuth2RestTemplateForUaaAdmin() {
         if (this.uaaAdminTemplate == null) {
             ClientCredentialsResourceDetails resource = new ClientCredentialsResourceDetails();
             resource.setAccessTokenUri(this.uaaTokenUrl);
             resource.setClientId("admin");
-            resource.setClientSecret("adminsecret");
+            resource.setClientSecret(this.uaaAdminSecret);
+            this.uaaAdminTemplate = new OAuth2RestTemplate(resource);
+            this.uaaAdminTemplate.setRequestFactory(new HttpComponentsClientHttpRequestFactory());
+        }
+
+        return this.uaaAdminTemplate;
+    }
+
+    public OAuth2RestTemplate getOAuth2RestTemplateForUaaAdmin(final String tokenUri) {
```

Comment:
Agreed. For now that means we will need to have a field with ```@Value``` in ACSAcceptanceIT as well as ACSRestTemplateFactory because other tests need other method ```getOAuth2RestTemplateForUaaAdmin``` with not parameters. This can be changed in the future story.